test/tests/orbit-common/unit/transaction-test.js

```diff
@@ -0,0 +1,134 @@
+
+///////////////////////////////////////////////////////////////////////////////
+
+var source;
+var schema;
+var transaction;
+
+module("OC - Transaction", {
+  setup: function() {
+    Orbit.Promise = Promise;
+
+    schema = new Schema({
+      models: {
+        planet: {
+          attributes: {
+            name: {type: 'string'},
+            classification: {type: 'string'}
+          },
+          links: {
+            moons: {type: 'hasMany', model: 'moon', inverse: 'planet'}
+          },
+          keys: {
+            id: { primaryKey: true, defaultValue: uuid }
+          }
+        },
+        moon: {
+          attributes: {
+            name: {type: 'string'}
+          },
+          links: {
+            planet: {type: 'hasOne', model: 'planet', inverse: 'moons'}
+          }
+        }
+      }
+    });
+
+    source = new MemorySource(schema);
+  },
+
+  teardown: function() {
+    schema = null;
+    source = null;
+    Orbit.Promise = null;
+  }
+});
+
+test("it exists", function(assert) {
+  var transaction = new Transaction(schema, {baseSource: source});
+  assert.ok(transaction);
+});
+
+test("requires the `baseSource` option", function(assert) {
+  assert.throws(function() {
+      var transaction = new Transaction(schema);
+    },
+    new Error("Assertion failed: `baseSource` must be supplied as an option when constructing a Transaction.")
+  );
+});
+
+test("automatically begins by default", function(assert) {
+  var transaction = new Transaction(schema, {baseSource: source});
+  assert.equal(transaction.active, true);
+});
+
+test("does not auto-begin if the `active` option = false", function(assert) {
+  var transaction = new Transaction(schema, {baseSource: source, active: false});
+  assert.equal(transaction.active, false);
+});
+
+test("once begun, tracks operations performed and inverse operations", function(assert) {
+  assert.expect(4);
+
+  var transaction = new Transaction(schema, {baseSource: source});
+
+  assert.equal(transaction.operations.length, 0, 'transaction has no operations');
+  assert.equal(transaction.inverseOperations.length, 0, 'transaction has no inverse operations');
+
+  return transaction.add('planet', {name: 'Jupiter', classification: 'gas giant'}).then(function(planet) {
+    equalOps(transaction.operations,
+      [{op: 'add', path: 'planet/' + planet.id, value: planet}],
+      'transaction tracked `add` operation');
+
+    equalOps(transaction.inverseOperations,
+      [{op: 'remove', path: 'planet/' + planet.id}],
+      'transaction tracked inverse operations');
+  });
+});
+
+test("`commit` applies operations to `baseSource`", function(assert) {
+  assert.expect(3);
+
+  var transaction = new Transaction(schema, {baseSource: source});
+
+  assert.equal(source.length('planet'), 0, 'base source has no planets');
+
+  return transaction.add('planet', {name: 'Jupiter', classification: 'gas giant'}).then(function(planet) {
+    return transaction.commit().then(function() {
```

Comment:
rather than chain `.then` inside the promise, `return transaction.commit();` and add the `.then()` to the chain?

Also, should we think about using a style pattern where we always break a newline on a `.then`?  I think this leads to less diff noise and easier readability.

Example:

```js
var savedPlanet;

return transaction.add('planet', {name: 'Jupiter', classification: 'gas giant'})
  .then(function(planet) {
    savedPlanet = planet;
    return transaction.commit();
  })
  .then(function() {
     assert....
     assert....
   });